src/librustc_mir/borrow_check/move_errors.rs

```diff
@@ -305,9 +314,29 @@ impl<'a, 'gcx, 'tcx> MirBorrowckCtxt<'a, 'gcx, 'tcx> {
 
                                 diag
                             }
-                            _ => self.infcx.tcx.cannot_move_out_of(
-                                span, "borrowed content", origin
-                            ),
+                            _ => {
+                                match self.borrowed_content_source(place) {
+                                    source @ BorrowedContentSource::Rc |
+                                    source @ BorrowedContentSource::Arc => {
+                                        let name = match source {
+                                            BorrowedContentSource::Rc => "Rc",
+                                            BorrowedContentSource::Arc => "Arc",
+                                            _ => unreachable!(),
+                                        };
+
+                                        let mut diag = self.infcx.tcx.cannot_move_out_of(
+                                            span, &format!("an `{}`", name), origin
+                                        );
+                                        diag.help(&format!(
+                                            "did you mean to call `{}::try_unwrap`?", name
```

Comment:
it seems pretty unlikely that they were intending to call `try_unwrap` but "accidentally" wrote `foo.bar`. Maybe we should word it more like: "you may wish to consider `{}::try_unwrap`"?

Reply:
I've updated the message.